src/test/rustdoc/description.rs

```diff
@@ -5,7 +5,7 @@
 //! It should not show up in the description.
 
 // @has 'foo/index.html' '//meta[@name="description"]/@content' \
-//   'Description test crate'
+//   ''
```

Comment:
Hmm maybe I should revert the hover text change because of this side effect.

Reply:
The problem is that this PR fixes some crates, like [once_cell](https://docs.rs/once_cell/1.8.0/once_cell/) (“Overview”), while making the one-line summary of [rocket](https://docs.rs/rocket/0.4.10/rocket/)’s crate worse (“Hi, and welcome to the core Rocket API documentation!”). The current behavior isn’t really a bug, and there isn’t a single, obviously-correct choice, so this seems like a lateral move, and not a clear-cut improvement.

While the hover-text might technically be generated by different code from the search summary, that’s mostly because the search summary can have code tags while the hover text cannot. Making one of them use headers while the other one skips headers seems bad. This is the same reason why I wouldn’t want the summary line selection algorithm to be different between crates, modules, or functions: it’s nice if the way summary lines are generated is easy-to-teach.

I almost want to “just make it configurable,” and add a way to explicitly choose the summary line, `#[doc(summary=“The Rust Standard Library”)]`. The upside is that if a crate author wants to start their crate docs with an “# Overview” section, they also have a way to write a good summary line. The downside is that explicitly-chosen summary lines can fall out of sync with reality, bitrotting into [metacrap](https://indieweb.org/metacrap).

Reply:
Yes, I see your point. One comment:

> making the one-line summary of rocket’s crate worse (“Hi, and welcome to the core Rocket API documentation!”)

Your point still holds, but just noting that the current summary is `Rocket - Core API Documentation`, and the new one would be empty (IIUC).

I think I will just close this PR for now since I don't think I'll have the time to investigate alternatives.

Reply:
> This is the same reason why I wouldn’t want the summary line selection algorithm to be different between crates, modules, or functions: it’s nice if the way summary lines are generated is easy-to-teach.

Notes for when we come back to this: There actually are differences in how people write docs for a crate, a mod, or a method. For a crate or mod, it's typical to start with a heading. For a method, it's rare. And the headings for methods are almost never summary-related; they are things like "Examples" and "Panics". Your point about teachability is well-made, but I think it's okay in this case to have special casing for methods vs everything else.